crates/mun_codegen/src/db.rs

```diff
@@ -14,6 +18,10 @@ pub trait IrDatabase: hir::HirDatabase {
     #[salsa::input]
     fn context(&self) -> Arc<Context>;
 
+    /// Returns the LLVM module that should be used for all generation steps.
+    #[salsa::input]
+    fn module(&self) -> Arc<Module>;
```

Comment:
This should definitely not be an input since its a transformation from HIR -> IR.

Reply:
I will resolve this issue in a future PR. I've changed the PR description to reflect this.